### DISCUSSION TITLE
How to assing role with custom guard name?

### DISCUSSION THREAD

**Discussion by alex8bits** [spatie/laravel-permission#2442] [Q&A]
Hi. If I get it right, assingRole() method allows to assign role only with default guard name. Is the a way to do it for roles with different guard names?

**Comment by parallels999** [comment]
Read documentation

**Comment by drbyte** [comment]
Lookup the role record which matches the desired guard, and assign that role record to the user:
```php
$role = Role::findByName('role name here', 'guard name here');
$user->assignRole($role);
```

**Reply by comhon-project** [reply]
if i'm not wrong, it will not work if the "different" guard name is not associated to model by methods listed in documentation, it will throw a `GuardDoesNotMatch` exception

**Comment by comhon-project** [comment]
you can assign roles with other guards than default guard ( see documentation [here](https://spatie.be/docs/laravel-permission/v5/basic-usage/multiple-guards) in the **note** section).
if the model is associated to the "different" guard name (with methods listed in documentation) the response of @drbyte should do the job .
but if it's not the case, for any reason, you can set  the `$guard_name` property on your model just before affect the role (with a different guard). 
```PHP
$role = Role::findByName('my_role', 'my_different_guard');
$user->guard_name = 'my_different_guard';
$user->assignRole($role);
```
but not sure if it's a good idea...
